apps/scheduler/src/services/scheduler_service.py

```diff
@@ -211,6 +211,14 @@ def _populate_match_schedule(self):
         4. Respect staggered matches configuration
         """
 
+         # Check if there are enough tables for ranking rounds
+        ranking_rounds = self.match_schedule[self.match_schedule["stage"] == "ranking"]["round"].nunique()
+        are_enough_tables = len(self.tables) >= ranking_rounds
```

Comment:
You can save a variable called min_matches_per_table which will be math.ceil(ranking_rounds/len(tables))

Then instead of checking whether the team was already on the table, check the number of times the teams was on the table and compare it to the min_matches_per_table